src/main/java/User.java

```diff
@@ -1,9 +1,42 @@
+import java.util.UUID;
+
 /**
  * Description: This class serves as a domain model for users.
  *
  * Intent: The User class represents user information, including the user's unique identifier, name, and password.
  * It provides methods to access and update user details.
  */
 public class User {
-    // TODO implement class
+
+    // Generated value
+    private final UUID id;
```

Comment:
There shouldn't be a need for this ID, since your ID is the username (as it's unique). 
So you can simplify by removing this and all the functions "getUserByID" etc..